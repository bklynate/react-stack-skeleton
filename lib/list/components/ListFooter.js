'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ListFooter;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The footer for the todo list.
 */
function ListFooter(props) {
  var activeItemsLeft = props.activeItemsLeft,
      filter = props.filter,
      onClickClearCompleted = props.onClickClearCompleted,
      otherProps = _objectWithoutProperties(props, ['activeItemsLeft', 'filter', 'onClickClearCompleted']);

  return _react2.default.createElement(
    'footer',
    { className: 'footer' },
    _react2.default.createElement(
      'span',
      { className: 'todo-count' },
      _react2.default.createElement(
        'strong',
        null,
        activeItemsLeft
      ),
      ' item left'
    ),
    _react2.default.createElement(
      'ul',
      { className: 'filters' },
      _react2.default.createElement(
        'li',
        null,
        _react2.default.createElement(
          _reactRouter.Link,
          { className: undefined /*filter? undefined: 'selected'*/, to: '/' },
          'All'
        )
      ),
      _react2.default.createElement(
        'li',
        null,
        _react2.default.createElement(
          _reactRouter.Link,
          { className: undefined /*filter === 'active'? 'selected': undefined*/, to: '/active' },
          'Active'
        )
      ),
      _react2.default.createElement(
        'li',
        null,
        _react2.default.createElement(
          _reactRouter.Link,
          { className: undefined /*filter === 'completed'? 'selected': undefined*/, to: '/completed' },
          'Completed'
        )
      )
    ),
    _react2.default.createElement(
      'button',
      { className: 'clear-completed', onClick: onClickClearCompleted },
      'Clear completed'
    )
  );
}