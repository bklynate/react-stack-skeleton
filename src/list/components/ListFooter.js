/* @flow */
import React from 'react';
import { Link } from 'react-router';

type Props = {
  activeItemsLeft: number,
  filter?: 'active' | 'completed',
  onClickClear: (e: SyntheticEvent) => void,
};

/**
 * The footer for the todo list.
 */
export default function ListFooter({ activeItemsLeft, filter, onClickClear, ...otherProps }: Props): React.Element<*> {
  return (
    <footer className="footer">
      <span className="todo-count"><strong>{activeItemsLeft}</strong> item left</span>
      <ul className="filters">
        <li>
          <Link className={filter? undefined: 'selected'} href="/">All</Link>
        </li>
        <li>
          <Link className={filter === 'active'? 'selected': undefined} href="/active">Active</Link>
        </li>
        <li>
          <Link className={filter === 'completed'? 'selected': undefined} href="/completed">Completed</Link>
        </li>
      </ul>
      <button className="clear-completed" onClick={onClickClear}>
        Clear completed
      </button>
    </footer>
  );
}
