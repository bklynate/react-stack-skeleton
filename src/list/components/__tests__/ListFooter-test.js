import React from 'react';
import renderer from 'react-test-renderer';

import ListFooter from '../ListFooter';

test('it displays the given number of remaining items.', () => {
  const component = renderer.create(
    <ListFooter activeItemsLeft={3} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays "all" link as selected when not filter is given.', () => {
  const component = renderer.create(
    <ListFooter />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays "active" link as selected when the filter is set to "active" given.', () => {
  const component = renderer.create(
    <ListFooter filter="active" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays "completed" link as selected when the filter is set to "completed" given.', () => {
  const component = renderer.create(
    <ListFooter filter="completed" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it calls the onClickClearCompleted function if the item is changed.', () => {
  const onClickFn = jest.fn();
  const component = renderer.create(
    <ListFooter onClickClearCompleted={onClickFn} />
  );
  const tree = component.toJSON();
  const clearButton = tree.children[2];

  clearButton.props.onClick();
  expect(onClickFn).toHaveBeenCalledTimes(1);
});
