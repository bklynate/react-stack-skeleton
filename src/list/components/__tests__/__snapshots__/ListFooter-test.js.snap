exports[`test it displays "active" link as selected when the filter is set to "active" given. 1`] = `
<footer
  className="footer">
  <span
    className="todo-count">
    <strong />
     item left
  </span>
  <ul
    className="filters">
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        All
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Active
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={undefined}>
    Clear completed
  </button>
</footer>
`;

exports[`test it displays "all" link as selected when not filter is given. 1`] = `
<footer
  className="footer">
  <span
    className="todo-count">
    <strong />
     item left
  </span>
  <ul
    className="filters">
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        All
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Active
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={undefined}>
    Clear completed
  </button>
</footer>
`;

exports[`test it displays "completed" link as selected when the filter is set to "completed" given. 1`] = `
<footer
  className="footer">
  <span
    className="todo-count">
    <strong />
     item left
  </span>
  <ul
    className="filters">
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        All
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Active
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={undefined}>
    Clear completed
  </button>
</footer>
`;

exports[`test it displays the given number of remaining items. 1`] = `
<footer
  className="footer">
  <span
    className="todo-count">
    <strong>
      3
    </strong>
     item left
  </span>
  <ul
    className="filters">
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        All
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Active
      </a>
    </li>
    <li>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={undefined}>
    Clear completed
  </button>
</footer>
`;
